cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactCodec.java

```diff
@@ -118,16 +127,50 @@ public FactCodec(EntityTable entityTable, int resolution, int rollTimebaseInterv
     return rowKey;
   }
 
+  /**
+   * 1) result row key length is determined by the tagValues to be included,
```

Comment:
you should put it into the comment inside the method and put javadoc description that I think I suggested